### PR TITLE
STORM-248 STORM-322 are fixed and upmerged

### PR DESCRIPTION
**Pull request by fmazoyer**
Details are in JIRA https://issues.apache.org/jira/browse/STORM-248.
As a collateral benefit, this also addresses JIRA https://issues.apache.org/jira/browse/STORM-322.
Could you please review?

### CODE REVIEW

File: bin/storm [hunk 1]
```diff
@@ -74,7 +74,8 @@ if (not os.path.isfile(os.path.join(USER_CONF_DIR, "storm.yaml"))):
 STORM_RELEASE_DIR = os.path.join(STORM_DIR, "RELEASE")
 STORM_LIB_DIR = os.path.join(STORM_DIR, "lib")
 STORM_BIN_DIR = os.path.join(STORM_DIR, "bin")
-STORM_LOGBACK_FILE_PATH = os.path.join(STORM_DIR, "logback", "cluster.xml")
+STORM_LOGBACK_DIR = os.path.join(STORM_DIR, "logback")
+STORM_LOGBACK_FILE_PATH = os.path.join(STORM_LOGBACK_DIR, "cluster.xml")
```
Comment (revans2) [inline bin/storm]: I don't think this is used any more.

**Comment by revans2** [comment]
One minor comment, but other then that it looks good.

**Comment by fmazoyer** [comment]
Thank you for your comment, this is done :-)
Could you please review ?

**Comment by revans2** [comment]
Looks good to me +1.  I'll give it a few days to let others have a chance to review, and then check it in if there are no other comments.

File: bin/storm [hunk 2]
```diff
@@ -312,6 +312,13 @@ def repl():
     cppaths = [CLUSTER_CONF_DIR]
     exec_storm_class("clojure.main", jvmtype="-client", extrajars=cppaths)
 
+def get_logback_conf_dir():
+    cppaths = [CLUSTER_CONF_DIR]
+    storm_logback_conf_dir = confvalue("storm.logback.conf.dir", cppaths)
+    if(storm_logback_conf_dir == None or storm_logback_conf_dir == "nil"):
```
Comment (harshach) [inline bin/storm]: Here storm_logback_conf_dir == "nil" will match if the storm_logback_conf_dir is set to  "nil" . I think you meant to check if the string is empty, a better way probably is "if not storm_logback_conf"

Comment (revans2) [inline bin/storm]: No it is actually "nil".  You can try it.

```
$ storm localconfvalue something.bogus
something.bogus: nil
```

Comment (revans2) [inline bin/storm]: If we want it to behave differently we need to fix confvalue.

Comment (harshach) [inline bin/storm]: Sorry, you are right it is "nil".

**Comment by harshach** [comment]
+1

**Comment by fmazoyer** [comment]
Hi Robert,
do you think there are still some things to do on my side ?
Thanks for everything :-)

**Comment by harshach** [comment]
@fmazoyer  Thanks for the patch. I pushed it into master.

**Comment by d2r** [comment]
@harshach, would you resolve STORM-248?  I Resolved STORM-322 already.

**Pull request merged**